src/CMakeLists.txt

```diff
@@ -27,6 +27,10 @@ if (WIN32)
 
 elseif (UNIX)
   add_executable(${EXECUTABLE_TITLE} ${SOURCE_FILES})
+  if(SCALEPROGBAR)
```

Comment:
Refactor this to check for the definition of the cache variable as defined as above. You will also need to add a bounds check. i.e. the value should not be negative